plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -512,6 +513,22 @@ protected TestTable createAggregationTestTable(String name, List<String> rows)
         return new TestTable(onRemoteDatabase(), name, "(short_decimal decimal(9, 3), long_decimal decimal(30, 10), t_double double precision, a_bigint bigint)", rows);
     }
 
+    @Test
+    public void testBigintAvgPushdown()
+    {
+        String schemaName = getSession().getSchema().orElseThrow();
+        // TODO (https://github.com/trinodb/trino/issues/5901) Use longer table name once Oracle version is updated
+        try (TestTable testTable = new TestTable(getQueryRunner()::execute, schemaName + ".avg_pushdown", "(t_bigint bigint)", ImmutableList.of("-1", "-2"))) {
+           // The output should be the same whether pushdown is occurring or not, but the plans should be different
+            if (hasBehavior(SUPPORTS_AGGREGATION_PUSHDOWN_AVG_BIGINT)) {
+                assertThat(query("SELECT avg(t_bigint) FROM " + testTable.getName())).matches("VALUES (DOUBLE'-1.5')").isFullyPushedDown();
```

Comment:
Good to know about `assertConditionallyPushedDown`. 

The matches bit actually caught a problem with the oracle connector returning a `decimal(19,0)` which rounds the output instead of a double. I'm digging into why other tests did not catch that problem

Reply:
Ok so trino's docs do not make this explicit, but `avg(x)` returns a double for all numeric input types *except* decimal, in which case it returns a decimal of the same precision/scale.
https://trino.io/docs/current/functions/aggregate.html#general-aggregate-functions

Trino's `bigint` type is translated to a `decimal(19,0)` in oracle, so when performing `avg(x)` over a bitint column you get a decimal output [in this case truncated to 0 scale] instead of a double.

So I guess that behavior is expected. It is weird though, because if `ImplementAvgBigint` was ever implemented for oracle the behavior would change (from returning a decimal(19,0) to a double)

Reply:
Ah, unintuitive indeed. Can you place a comment where we add the ImplementAvgBigint rewrite rule (or a better place if you know) in OracleClient? I'll (and others) will forget this eventually.

Reply:
The trino oracleclient does not implement `ImplementAvgBigint` because `bigint` does not exist in oracle. But I'll add a comment in the oracleclient around the typemappings section warning others.